src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/w3c/harvesters/JsElements.kt

```diff
@@ -0,0 +1,43 @@
+package com.atlassian.performance.tools.jiraactions.api.w3c.harvesters
+
+import com.atlassian.performance.tools.jiraactions.api.w3c.PerformanceElementTiming
+import org.openqa.selenium.JavascriptExecutor
+import org.openqa.selenium.TimeoutException
+
+
+internal fun getJsElementsPerformance(javascript: JavascriptExecutor): List<PerformanceElementTiming> {
+    return try {
+        val jsElements =
+            javascript.executeAsyncScript("new PerformanceObserver(list => arguments[0](list.getEntries())).observe({type: 'element', buffered: true})")
+        parseElements(jsElements)
+    } catch (timeoutEx: TimeoutException) {
+        // No element performance entries on page.
```

Comment:
Is there a way to detect lack of element timings without waiting for a timeout?

Reply:
Note that `RecordedPerformanceEntries` (including its `resources` field) is taken as a snapshot at the end of a JPT `Action`.
Recording of the performance entries does not wait for anything to load anymore. The JPT Action did the waiting already.

- [ ] avoid waiting in `JavascriptW3cPerformanceTimeline.record`

Reply:
Yep, seems like this is possible with the `.takeRecords()` method ([doc here](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/takeRecords)).
Testing in progress...